photutils/psf/models.py

```diff
@@ -4,17 +4,428 @@
 """
 
 from __future__ import division
+import warnings
 import numpy as np
+import copy
 from astropy.table import Table
 from astropy.modeling import models, Parameter, Fittable2DModel
 from astropy.modeling.fitting import LevMarLSQFitter
 from astropy.nddata.utils import subpixel_indices
 from ..utils import mask_to_mirrored_num
 from ..extern.nddata_compat import extract_array
 
+__all__ = ['FittableImageModel2D', 'NonNormalizable',
+           'IntegratedGaussianPRF', 'PRFAdapter',
+           'prepare_psf_model', 'get_grouped_psf_model']
 
-__all__ = ['IntegratedGaussianPRF', 'PRFAdapter', 'prepare_psf_model',
-           'get_grouped_psf_model']
+
+class NonNormalizable(Warning):
+    """
+    Used to undicate that a :py:class:`FittableImageModel2D` model is
+    non-normalizable.
+
+    """
+    pass
+
+
+class FittableImageModel2D(Fittable2DModel):
+    """
+    A fittable 2D model of an image allowing for image intensity scaling
+    and image translations.
+
+    This class takes 2D image data and computes the
+    values of the model at arbitrary locations (including at intra-pixel,
+    fractional positions) within this image using spline interpolation
+    provided by :py:class:`~scipy.interpolate.RectBivariateSpline`.
+
+    The fittable model provided by this class has three model parameters:
+    an image intensity scaling factor (`flux`) which is applied to
+    (normalized) image, and two positional parameters (`x_0` and `y_0`)
+    indicating the location of a feature in the coordinate grid on which
+    the model is to be evaluated.
+
+    If this class is initialized with `flux` (intensity scaling factor)
+    set to `None`, then `flux` is be estimated as ``|sum(data)|``.
```

Comment:
I found this default to be rather confusing.  This seems to "cancel out" the normalizing if you do `normalize=True`.  I would instead suggest that the default flux be `1` (consistent with the other PSF models), and have `normalize` default to True.  

It's also a bit problematic for model machinery that might assume that `FittableImageModel2D.flux.default` is a valid value to set the model back to (which it isn't for `None`, but is for `1`).